File: trip-service/src/main/java/com/study/trip/application/dto/TripResponse.java
package com.study.trip.application.dto;

import com.study.trip.domain.entity.Trip;
import com.study.trip.domain.entity.TripStatus;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Builder
public class TripResponse {
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;
    private TripStatus status;
    private LocalDateTime createdAt;

    public static TripResponse from(Trip trip) {
        return TripResponse.builder()
                .id(trip.getId())
                .userId(trip.getUserId())
                .title(trip.getTitle())
                .description(trip.getDescription())
                .startDate(trip.getStartDate())
                .endDate(trip.getEndDate())
                .status(trip.getStatus())
                .createdAt(trip.getCreatedAt())
                .build();
    }
}

File: trip-service/src/main/kotlin/com/study/trip/application/dto/TripResponse.kt
package com.study.trip.application.dto

import com.study.trip.domain.entity.Trip
import com.study.trip.domain.entity.TripStatus
import java.time.LocalDate
import java.time.LocalDateTime

data class TripResponse(
    val id: Long,
    val userId: Long,
    val title: String,
    val description: String?,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val status: TripStatus,
    val createdAt: LocalDateTime
) {
    companion object {
        fun from(trip: Trip) = TripResponse(
            id = trip.id, userId = trip.userId, title = trip.title,
            description = trip.description, startDate = trip.startDate,
            endDate = trip.endDate, status = trip.status, createdAt = trip.createdAt
        )
    }
}
